feldman_lab_to_nwb/convert_feldman/feldman_electrode_widget.py

```diff
@@ -1,51 +1,183 @@
 import numpy as np
+from typing import Optional
 
 from ipywidgets import widgets
+from pynwb import NWBFile
 from pynwb.misc import Units
 from nwbwidgets import default_neurodata_vis_spec
 from nwbwidgets.misc import PSTHWidget
 from nwbwidgets import base
 import plotly.graph_objects as go
 
+ELECTRODE_COLOR = "#000000"
 
-BACKGROUND_COLOR = "#9FE19D"
-ELECTRODE_COLOR = "#d81700"
-DETECTED_COLOR = "#002bd8"
-SELECTED_COLOR = "#fccb00"
+ELECTRODE_SIZE = 3
+DETECTED_SIZE = 10
+SELECTED_SIZE = 20
+
+
+def calculate_response(
```

Comment:
@CodyCBakerPhD Could you make the "responsiveness calculation done in a function on a per-unit basis and call it in a loop? I think that way would be easier to test and to follow.

Reply:
@bendichter I've broken up the main code to operate in more relevant pieces on a per-unit basis. The primary `calculate_response` function still includes the iteration over units, but passes on the lifting down to the other helper functions in the new utils file.

I'll go ahead and share the NWBFile I'm working with on Slack as well; to get the full internal variables returned for inspection simply run

```
with NWBHDF5IO(path=nwbfile_name, mode="r") as io:
    nwbfile = io.read()
    response, internals = calculate_response(units=nwbfile.units, trials=nwbfile.trials, pre_alignment_window=[0, 0.5], evoked_window=[0.1, 0.15], event_name="start_time")
```

Reply:
For easy reference, to see it from the widget view with this branch checked out,
```
from feldman_lab_to_nwb import nwb2widget
from pynwb import NWBHDF5IO


nwbfile_name = ".../rapid_testing_LR_210209_g0_full.nwb"

io = NWBHDF5IO(nwbfile_name, "r")
nwbfile=io.read()
nwb2widget(nwbfile)
```
to which I have now added a temporary box for setting the offset for evoked responsitivity calculation. I've also added the raw responsitivity value for the channels as text to the channel hover box on the grid.

Reply:
Temporary box removed, now coupled with PSTH view.